components/ca-resource.html

```diff
@@ -0,0 +1,123 @@
+<script>
+    define('ca-resource', [], () => {
+        /**
+         * Non-UI component which declares a remote resource accessed through HTTP. The value of the resource is
+         * exposed through the {@link value} property. The following events are fired by this element:
+         *
+         * <dl>
+         * <dt>ca-resource-fetching</dt><dd>Fired when the resource is about to be fetched, with the fetch options available to be modified.</dd>
+         * <dt>ca-resource-fetched</dt><dd>Fired when the resource has been fetched, with an error or non-error result.</dd>
+         * <dt>ca-resource-ready</dt><dd>Fired when the resource has been fetched successfully. {@link value} will be available.
+         * <dt>ca-resource-unauthorized</dt><dd>Fired when the resource fetch failed with 401 Unauthorised because the user's credentials could not be verified.</dd>
+         * <dt>ca-resource-forbidden</dt><dd>Fired when the resource fetch failed with 403 Forbidden because the user's credentials were not acceptable to retrieve it.</dd>
+         * </dl>
+         * TODO: make this able to PUT resources back by changing the 'value' property
+         */
+        class ApiResourceElement extends HTMLElement {
+            /**
+             * observedAttributes - get a list of attributes that should effect re-fetch.
+             * @return {Array} list of attributes that effect a re-fetch.
+             */
+            static get observedAttributes() {
+                return ['href'];
+            }
+
+            /**
+             * @description Executes when any attribute is changed on the element
+             * @type {Event}
+             * @param {string} attrName - the name of the attribute to have changed
+             * @param {string} oldVal - the old value of the attribute
+             * @param {string} newVal - the new value of the attribute
+             * @returns {undefined}
+             */
+            attributeChangedCallback(attrName, oldVal, newVal) {
+                if (ApiResourceElement.observedAttributes.indexOf(attrName) >= 0) {
```

Comment:
could this use ApiResourceElement.observedAttributes.includes(attrName)?

Reply:
Yep